DirectX11Proj/modelclass.h

```diff
@@ -8,41 +8,34 @@ using namespace DirectX;
 #include <memory>
 
 class d3dVertexBuffer;
-
+class d3dTexture;
 struct MeshData;
 
 class ModelClass
 {
-private:
-	struct VertexType
-	{
-		XMFLOAT3 position;
-		XMFLOAT4 color;
-	};
-
 public:
 	ModelClass();
 	ModelClass(const ModelClass&);
 	~ModelClass();
 
-	bool Initialize(ID3D11Device*);
 	bool Initialize(ID3D11Device*, const MeshData&);
 
 	void Shutdown();
 	void Render(ID3D11DeviceContext*);
 
 	int GetIndexCount();
 
-private:
-	bool InitializeBuffers(ID3D11Device*);
-	bool InitializeBuffers(ID3D11Device*, const MeshData&);
 
-	// Initialize buffers using previously loaded data
+	d3dTexture* mpTexture;
+	bool mHastexture = false;
```

Comment:
make this private and make a get:
const bool HasTexture() const;

or
const uint8_t GetTextureCount() const;